enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/CloudAssetsJavaCompletion.java

```diff
@@ -106,8 +107,12 @@ private Class<? extends OCIItem> element2ReferenceClass(Element element) {
             Elements elements = this.processingEnv.get().getElementUtils();
             Types types = this.processingEnv.get().getTypeUtils();
             tm = types.erasure(tm);
-            TypeElement dataSource = elements.getTypeElement(DATA_SOURCE);
-            if (dataSource != null && types.isAssignable(tm, dataSource.asType())) {
+            TypeElement javaDataSource = elements.getTypeElement(JAVA_DATA_SOURCE);
```

Comment:
Please double check this `java.sql.DataSource` type is valid -- I haven't found it in JDK17-21 javadocs. There's just `javax.sql.DataSource` (although in `java.sql` module) ...

Reply:
Fixed. Good catch.